skrub/_sentence_encoder.py

```diff
@@ -0,0 +1,300 @@
+        accross embedding models, this class uses a :class:`~sklearn.decomposition.PCA`
+        to set the number of embedding to ``n_components`` during ``transform``.
+        Set ``n_components=None`` to skip the PCA dimension reduction mechanism.
+
+        See [1]_ for more details on the choice of the PCA and default ``n_components``.
+
+    device : str, default=None
+        Device (e.g. "cpu", "cuda", "mps") that should be used for computation. If None,
+        checks if a GPU can be used.
+        Note that macOS ARM64 users can enable the GPU on their local machine
+        by setting ``device="mps"``.
+
+    batch_size : int, default=32
+        The batch size to use during ``transform``.
+
+    use_auth_token : bool or str, default=None
+        HuggingFace authentication token to download private models.
+
+    cache_folder : str, default=None
+        Path to store models. By default ``~/skrub_data``.
+        See :func:`skrub.datasets._utils.get_data_dir`.
+
+    random_state : int, RandomState instance or None, default=None
+        Used when the PCA dimension reduction mechanism is used, for reproducible
+        results across multiple function calls.
+
+    verbose : bool, default=True
+        Verbose level, controls whether to show a progress bar or not during
+        ``transform``.
+
+    References
+    ----------
+    .. [1]  L. Grinsztajn, M. Kim, E. Oyallon, G. Varoquaux
+            "Vectorizing string entries for data processing on tables: when are larger
+            language models better?", 2023.
+            https://hal.science/hal-04345931
+    """
+
+    def __init__(
+        self,
+        model_name="intfloat/e5-small-v2",
+        n_components=30,
+        device=None,
+        batch_size=32,
+        use_auth_token=None,
+        cache_folder=None,
+        random_state=None,
+        verbose=False,
+    ):
+        self.model_name = model_name
+        self.n_components = n_components
+        self.device = device
+        self.batch_size = batch_size
+        self.use_auth_token = use_auth_token
+        self.cache_folder = cache_folder
+        self.random_state = random_state
+        self.verbose = verbose
+
+    def fit_transform(self, column, y=None):
+        """Fit the SentenceEncoder from ``column``.
+
+        In practice, it loads the pre-trained model from disk and returns
+        the embeddings of the column.
+
+        Parameters
+        ----------
+        column : pandas or polars Series of shape (n_samples,)
+            The string column to compute embeddings from.
+
+        y : None
+            Unused. Here for compatibility with scikit-learn.
+
+        Returns
+        -------
+        X_out : pandas or polars DataFrame of shape (n_samples, n_components)
+            The embedding representation of the input.
+        """
+        st = import_optional_dependency("sentence_transformers")
+
+        if not sbd.is_string(column):
```

Comment:
not important but if we do the check before we avoid importing the library in the case where we won't use it and raise an error anyway. but it's true that we usually put imports at the top

Reply:
Doesn't cost too much to change :)